gen/xbuild.sh

```diff
@@ -19,69 +19,69 @@ mkdir dist || exit 5
 ERRORS=
 
 build() {
-    echo "=== Building for GOOS=$1 and GOARCH=$2."
     # https://golang.org/doc/install/source#environment
+    echo "=== Building for GOOS=$1 and GOARCH=$2."
     CGO_ENABLED=0 GOOS="$1" GOARCH="$2" go build -o dist/ \
         -ldflags="-s -w -X main.gVersion=$version"
     if test "$?" != "0"; then
         ERRORS=1
     else
-        case "$3" in
-        *.tar.gz)
-            (
-                cd dist
-                tar czf "$3" "$4" --remove-files
-            ) || exit 5
-            ;;
-        *.zip)
-            (
-                cd dist
-                zip "$3" "$4" --move
-            ) || exit 5
-            ;;
-        esac
-        echo "dist/$3 successfully created."
+        (
```

Comment:
This is just a suggestion, but if you're worried about nesting/complexity, you could extract this out into a separate function:

```shell
package() (
    cd dist || return 1
    ...
    do_something || return 1
)
```

TIL that if you use `( )` instead of `{ }`, the function will run in a separate subshell so the effect of `cd` will be restored.

You could also similar extract out the `go build` part into a separate `build` command and then write a high-level function (e.g. `process`) that invokes both of them.

Reply:
Good idea. I only did the `process` part, though.

> TIL that if you use ( ) instead of { }, the function will run in a separate subshell so the effect of cd will be restored.

TIL as well...